src/js/actions/libraries.js

```diff
@@ -106,13 +111,210 @@ define(function (require, exports) {
+     */
+    var createLayerStyleFromSelectedLayer = function () {
+        var appStore = this.flux.store("application"),
+            libStore = this.flux.store("library"),
+            currentDocument = appStore.getCurrentDocument(),
+            currentLibrary = libStore.getCurrentLibrary(),
+            currentLayers = currentDocument.layers.selected;
+            
+        if (!currentLibrary || currentLayers.count() !== 1) {
+            return Promise.resolve();
+        }
+
+        // FIXME: All constants like this should be described in one location for later
+        var LAYERSTYLE_TYPE = "application/vnd.adobe.element.layerstyle+dcx",
+            REPRESENTATION_TYPE = "application/vnd.adobe.layerstyle";
+
+        var currentLayer = currentLayers.first(),
+            layerRef = layerEffectAdapter.referenceBy.id(currentLayer.id),
+            stylePath = "/tmp/layerStyleExport.asl",
+            thumbnailPath = "/tmp/layerStyleThumbnail.png",
+            saveLayerStyleObj = layerEffectAdapter.saveLayerStyleFile(layerRef, stylePath, thumbnailPath);
+
+        currentLibrary.beginOperation();
+        
+        // Create the layer style asset        
+        var newElement = currentLibrary.createElement(currentLayer.name, LAYERSTYLE_TYPE);
+        
+        // Then, have PS generate the style file (.asl) and the thumbnail (.png)
+        return descriptor.playObject(saveLayerStyleObj)
+            .bind(this)
+            .then(function () {
+                // Assign it's primary representation (.asl file)
+                var representation = newElement.createRepresentation(REPRESENTATION_TYPE, "primary");
+
+                return Promise.fromNode(function (cb) {
+                    representation.updateContentFromPath(stylePath, false, cb);
+                });
+            }).then(function () {
+                // Assign the thumbnail to rendition
+                var rendition = newElement.createRepresentation("image/png", "rendition");
+
+                return Promise.fromNode(function (cb) {
+                    rendition.updateContentFromPath(thumbnailPath, false, cb);
+                });
+            }).then(function () {
+                // FIXME: Constant here
+                newElement.setRenditionCache(108, thumbnailPath, function () {
+                    // FIXME: In CEP panel, they delete the temporary thumbnail file afterwards
+                });
+            }).then(function () {
+                currentLibrary.endOperation();
+                // FIXME: Do we need payload?
+                return this.dispatchAsync(events.libraries.ASSET_CREATED, {});
+            });
+    };
+    createLayerStyleFromSelectedLayer.reads = [locks.JS_DOC, locks.JS_LIBRARIES, locks.CC_LIBRARIES];
+    createLayerStyleFromSelectedLayer.writes = [locks.JS_LIBRARIES];
+
+    /**
+     * Uploads the given color as a color asset
+     *
+     * @param {{r: number, g: number, b: number}} color Color to be added as an asset
```

Comment:
If this param is a `Color` model, it might be nicer to just include a `toHexString()` method so that wraps the `tinycolor` dependency. Also, the `@param` should just be `{Color}`.